### PR TITLE
chore: updated packages, fixed build bugs and migrated packages

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

- Updates all packages to their latest minor versions as safe as possible
- Upgraded to Sentry 8x (and migrated) package for Next.js, introducing usage of Next.js's Telemetry module and OpenTelemtry
- Updated Shiki usage to import the main package directly whenever possible and force Next.js to do optimized module resolution for the 'shiki' package
- Disabled `--turbo` for the time being, which reduces local development speed, but otherwise we can't update to Next.js 14.2

## Validation

Local development, build and deployments should work as expected

### CODE REVIEW

**Review by mikeesto — APPROVED** [review]
LGTM. I will an open an issue to track progress on re-enabling the turbo flag (I saw you reached out to folks at Vercel).

**Review by aymen94 — APPROVED** [review]

**Comment by ovflowd** [comment]
@MattIPv4 could you give a review? We need Web Infra to review this PR 🙇

**Comment by bmuenzenmeyer** [comment]
I have this error on a clean pull of the repo - investigating

image

**Comment by ovflowd** [comment]
> I have this error on a clean pull of the repo - investigating
> 
> image

Is it a clean pull? It says there is already another process running. These DOCTYPE issues generally occur when we try to fetch Node.js data, but the server fails. (Like the releases, api posts, etc)... But LMK if you find anything!

File: next.config.mjs [hunk 1]
```diff
@@ -100,9 +109,12 @@ const nextConfig = {
       '@radix-ui/react-select',
       '@radix-ui/react-toast',
       'tailwindcss',
+      'shiki',
     ],
     // Removes the warning regarding the WebPack Build Worker
     webpackBuildWorker: false,
+    // Enables Next.js's Instrumentation Hook
+    instrumentationHook: true,
```
Comment (bmuenzenmeyer) [inline next.config.mjs:117]: this is neat https://nextjs.org/docs/app/api-reference/next-config-js/instrumentationHook

**Review by bmuenzenmeyer — APPROVED** [review]
works locally

**Comment by bmuenzenmeyer** [comment]
@ovflowd 3000 was another app running, not nodejs.org - i killed it and 3000 works fine

File: next.config.mjs (hunk 1 above)
Comment (ovflowd) [reply next.config.mjs:117]: It is! Had no idea this existed!

**Comment by MattIPv4** [comment]
> @MattIPv4 could you give a review? We need Web Infra to review this PR 🙇 

Sorry for the slow response, am hiking around New Zealand currently. Looks like this was merged okay?

**Pull request merged**